### PR TITLE
added application/merge-patch+json

### PR DESCRIPTION
**Pull request by Executioner1939**
added `application/merge-patch+json` media type to akka-http-core
Refs #2189

### CODE REVIEW

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/MediaType.scala [hunk 1]
```diff
@@ -336,6 +336,7 @@ object MediaTypes extends ObjectRegistry[(String, String), MediaType] {
   val `application/javascript`                                                    = awoc("javascript", "js")
   val `application/json`                                                          = awfc("json", HttpCharsets.`UTF-8`, "json")
   val `application/json-patch+json`                                               = awfc("json-patch+json", HttpCharsets.`UTF-8`)
+  val `application/merge-patch+json`                                              = awfc("merge-patch+json", HttpCharsets.`UTF-8`)
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/MediaType.scala]: please mirror this in javadsl

**Comment by ktoso** [comment]
Please refer to ticket this addresses in comment, not title (in title it does not link properly)

Refs https://github.com/akka/akka-http/issues/2189

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/MediaType.scala (hunk 1 above)
Comment (Executioner1939) [reply akka-http-core/src/main/scala/akka/http/scaladsl/model/MediaType.scala]: Hi @ktoso I have added it to the JavaDSL, apologies for missing that :-).

**Comment by jrudolph** [comment]
OK TO TEST

**Pull request merged**